pkg/asset/installconfig/aws/validation_test.go

```diff
@@ -155,6 +156,204 @@ func TestValidate(t *testing.T) {
+			images: map[string]*ImageInfo{
+				"ami-uefi": {BootMode: "uefi"},
+			},
+		},
+		{
+			name: "valid instance type with enabling SEV-SNP on compute",
+			installConfig: icBuild.build(
+				icBuild.withInstanceType("m5.xlarge", "m5.xlarge", "m6a.xlarge"),
+				icBuild.withComputeCPUOptions(&aws.CPUOptions{
+					ConfidentialCompute: ptr.To(aws.ConfidentialComputePolicySEVSNP),
+				}, 0),
+				icBuild.withComputePlatformAMI("ami-uefi", 0),
+			),
+			availRegions:  validAvailRegions(),
+			availZones:    validAvailZones(),
+			instanceTypes: validInstanceTypes(),
+			images: map[string]*ImageInfo{
+				"ami-uefi": {BootMode: "uefi"},
+			},
+		},
+		{
+			name: "invalid instance type with enabling SEV-SNP on control plane",
+			installConfig: icBuild.build(
+				icBuild.withInstanceType("m6a.xlarge", "m5.xlarge", "m6a.xlarge"),
+				icBuild.withControlPlaneCPUOptions(&aws.CPUOptions{
+					ConfidentialCompute: ptr.To(aws.ConfidentialComputePolicySEVSNP),
+				}),
+				icBuild.withControlPlanePlatformAMI("ami-uefi"),
+			),
+			availRegions:  validAvailRegions(),
+			availZones:    validAvailZones(),
+			instanceTypes: validInstanceTypes(),
+			images: map[string]*ImageInfo{
+				"ami-uefi": {BootMode: "uefi"},
+			},
+			expectErr: `^\QcontrolPlane.platform.aws.type: Invalid value: "m5.xlarge": specified instance type in the specified region doesn't support amd-sev-snp\E$`,
+		},
+		{
+			name: "invalid instance type with enabling SEV-SNP on compute",
+			installConfig: icBuild.build(
+				icBuild.withInstanceType("m6a.xlarge", "m6a.xlarge", "m5.xlarge"),
+				icBuild.withComputeCPUOptions(&aws.CPUOptions{
+					ConfidentialCompute: ptr.To(aws.ConfidentialComputePolicySEVSNP),
+				}, 0),
+				icBuild.withComputePlatformAMI("ami-uefi", 0),
+			),
+			availRegions:  validAvailRegions(),
+			availZones:    validAvailZones(),
+			instanceTypes: validInstanceTypes(),
+			images: map[string]*ImageInfo{
+				"ami-uefi": {BootMode: "uefi"},
+			},
+			expectErr: `^\Qcompute[0].platform.aws.type: Invalid value: "m5.xlarge": specified instance type in the specified region doesn't support amd-sev-snp\E$`,
+		},
+		{
+			name: "valid instance type with no confidential compute settings",
+			installConfig: icBuild.build(
+				icBuild.withInstanceType("m5.xlarge", "m5.xlarge", "m5.xlarge"),
+			),
+			availRegions:  validAvailRegions(),
+			availZones:    validAvailZones(),
+			instanceTypes: validInstanceTypes(),
+		},
+		{
+			name: "valid instance type with disabling confidential compute",
+			installConfig: icBuild.build(
+				icBuild.withInstanceType("m5.xlarge", "m5.xlarge", "m5.xlarge"),
+				icBuild.withControlPlaneCPUOptions(&aws.CPUOptions{
+					ConfidentialCompute: ptr.To(aws.ConfidentialComputePolicyDisabled),
+				}),
+			),
+			availRegions:  validAvailRegions(),
+			availZones:    validAvailZones(),
+			instanceTypes: validInstanceTypes(),
+		},
+		{
+			name: "valid UEFI AMI with enabling SEV-SNP on control plane",
```

Comment:
This case has the same setting as test case `valid instance type with enabling SEV-SNP on control plane`, right? We can combine the 2 cases into:

```
valid instance type with enabling SEV-SNP and UEFI AMI on control plane
```

Test cases for compute pool also has duplicates :eyes: